src/mlpack/methods/amf/init_rules/average_init.hpp

```diff
@@ -74,6 +74,77 @@ class AverageInitialization
     H = H + avgV;
   }
 
+  /**
+   * Initialize the matrix W or H to the average value of V with uniform
+   * random noise added.
+   *
+   * @param V Input matrix.
+   * @param r Rank of matrix.
+   * @param M W or H matrix, to be initialized to the average value of V
+   * with uniform random noise added.
+   * @param whichMatrix If true, initialize W. Otherwise, initialize H.
+   */
+  template<typename MatType>
+  inline static void InitializeOne(const MatType& V,
+                                   const size_t r,
+                                   arma::mat& M,
+                                   const bool whichMatrix = true)
+  {
+    const size_t n = V.n_rows;
+    const size_t m = V.n_cols;
+
+    if (whichMatrix)
+    {
+      double avgV = 0;
+      size_t count = 0;
+      double min = DBL_MAX;
+
+      // Iterate over all elements in the matrix (for sparse matrices, this only
+      // iterates over nonzeros).
+      for (typename MatType::const_row_col_iterator it = V.begin();
+          it != V.end(); ++it)
+      {
+        ++count;
```

Comment:
Looks like this parameter isn't used, maybe I missed something?

Reply:
Yeah I noticed that. This parameter seems to be redundant since we do not have to count the number of elements in matrix V. It's just n * m.

Reply:
Right, fine to remove it.